### ISSUE TITLE
Workflow for getting the job status (can we do blocking request?)

### ISSUE THREAD

**Issue by grzegorznowak** [jenkinsci/generic-webhook-trigger-plugin#71]
Hey, g'day man,

my question is how would you go about receiving a status from a triggered job ? Can a request be made blocking somehow until  a job/pipeline has finished or would you use another API call to jenkins (and while we do get queued item's id in return it should be quite possible I guess, like queue/item/182263/) to pool for that information.
I think having a convenience of making request blocking, and at least getting back the build status, would be quite cool. Where should I be looking in codes in order to get a hook point for a PR if that feature is not there already ? Much appreciated.

**Comment by tomasbjerre** [comment]
The plugin has no feature that would hang on to the request until the triggered jobs are done.

I would (1) let the pipeline perform a callback to the invoking system to report that the job is done. Or (2) poll any jobs until they are completed.

Here is where the job gets triggered: https://github.com/jenkinsci/generic-webhook-trigger-plugin/blob/8488a932b9b5b7e84567948ed616e82d5b0b72f2/src/main/java/org/jenkinsci/plugins/gwt/GenericTrigger.java#L132

**Comment by grzegorznowak** [comment]
Yeah I kind of can't setup a listen server, since it's a part of unit testing suite, so don't want to complicate it further with stub servers and such. A one-off blocking request would be sooo nice. Polling is an option, and might go with it in the end. 
But have found that place just now too, and looking at jenkins's API one could make one extra call to `.getFuture()` method of the `QueueItem` model and that should be that, no ?

**Comment by tomasbjerre** [comment]
Yes. `.getFuture().get()` should do it.

I'm thinking this should be a feature that is enabled by a special request and not a configuration. Perhaps this can be enabled by a query parameter like `wait_for_completion=true`.

Another issue is how to wait. The jobs are now found and [triggered one by one](https://github.com/jenkinsci/generic-webhook-trigger-plugin/blob/8488a932b9b5b7e84567948ed616e82d5b0b72f2/src/main/java/org/jenkinsci/plugins/gwt/GenericWebHookRequestReceiver.java#L106). I'm thinking this feature will do that also. Trigger one job, wait for it, trigger next job, wait for that.... If one would want to trigger several jobs and have them run in parallel, a simple pipeline job could be created to orchestrate that. Let the plugin only trigger that pipeline job and have it trigger other jobs in parallell.

What do you think?

**Comment by grzegorznowak** [comment]
Dang ok, that native runner itself only triggers a queue task. Back to the drawing board then :/

**Comment by tomasbjerre** [comment]
You should not need to pass the username/password `https://{}:{}@...`.

Thanks for posting. Hopefully someone else will find it useful.

**Comment by grzegorznowak** [comment]
oh yeah thanks, in the first one we use the token sure. The other two are core API calls so I believe it's the only way it will work (?)

**Comment by tomasbjerre** [comment]
Yes correct.

**Comment by alexandrosntak** [comment]
Hello @tomasbjerre . I tried your solution in the feature/wait branch and it works, thanks. 👍 

I have two questions.

1. Do you have an approximate date of release for this feature?
2. Is there a way to insert data into the response?

**Comment by tomasbjerre** [comment]
1. No, I'm not sure it is possible.
2. No. It will return some information in the response by default. And that information can optionally be made silent. If you want some specific information in the response, open a new issue about that.
